### PR TITLE
Add remote sampling README instructions - Golang

### PR DESCRIPTION
**Pull request by bhautikpip**
This PR introduces remote sampling instructions for go. We can merge this PR after this [PR changes](https://github.com/open-telemetry/opentelemetry-go-contrib/pull/1859) has been  merged and released.

### CODE REVIEW

File: src/docs/getting-started/go-sdk/trace-manual-instr.mdx [hunk 1]
```diff
@@ -169,6 +169,34 @@ if err != nil {
 
 <SectionSeparator />
 
+## Using X-Ray Remote Sampling
```
[the remark this answers is not in the document]
Comment (bhautikpip) [reply src/docs/getting-started/go-sdk/trace-manual-instr.mdx:172]: Yeah we definitely can but I'm not sure if `traceIDRationBasedSampler` would come in the ADOT documentation scope. Those samplers would already be included in the upstream OTel documentation like [this](https://opentelemetry-python.readthedocs.io/en/latest/sdk/trace.sampling.html). Maybe here we can just point them to this documentation if the goal here is to just customer make aware about the usage of `traceIDRationBasedSampler`.

File: src/docs/getting-started/go-sdk/trace-manual-instr.mdx [hunk 2]
```diff
@@ -169,6 +169,34 @@ if err != nil {
...
+
+AWS X-Ray remote sampler can be initialized using `NewRemoteSampler` API. `NewRemoteSampler` can be configured with below options.
+
+| **NewRemoteSampler API options **                      | **Description**                                                      | **Default Value**       |
+|--------------------------------------------------------|----------------------------------------------------------------------|-------------------------|
+| WithEndpoint(endpoint url.URL)                         | Endpoint used to communicate with the `awsproxy` collector extension | http://localhost:2000   |
+| WithSamplingRulesPollingInterval(time.Duration)        | Duration between polling of the GetSamplingRules API                 | 300 seconds             |
+| WithLogger(l logr.Logger)                              | logging for remote sampler                                           | stdr                    |
```
[the remark this answers is not in the document]
Comment (bhautikpip) [reply src/docs/getting-started/go-sdk/trace-manual-instr.mdx]: basically this: https://github.com/go-logr/stdr

File: src/docs/getting-started/go-sdk/trace-manual-instr.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (willarmiros) [reply src/docs/getting-started/go-sdk/trace-manual-instr.mdx:172]: Yeah it's definitely a valid point - I initially thought that it wasn't in scope which is why it wasn't in the initial templates. However after working with some customers who had extremely high billing due to the 100% sampling of OTel, I thought calling out the default sampling behavior was important so I added it [to the template](https://github.com/willarmiros/aws-otel-docs/blob/main/templates/manual-trace-instrumentation.md#configuring-sampling). I think it's important to mention because it's OTel's equivalent of "local sampling"

Basically I think we should just have a small separate section about sampling in general that:
* Call out that sampling is 100% by default
* Link to the upstream trace ID ratio docs to indicate this is the upstream way of reducing sampling rate

[the remark this answers is not in the document]
Comment (bhautikpip) [reply src/docs/getting-started/go-sdk/trace-manual-instr.mdx:172]: Yeah I think it's fair to clear out what you suggested if this will improve customer experience.

**Review by willarmiros — APPROVED** [review]
lgtm once the CI is passing!

File: src/docs/getting-started/go-sdk/trace-manual-instr.mdx [hunk 3]
```diff
@@ -169,6 +169,52 @@ if err != nil {
...
+
+```go lineNumbers=true
+   ctx := context.Background()
+
+   endpoint, err := url.Parse("http://127.0.0.1:2000"); if err != nil {
+         return
+   }
+
+   // instantiate remote sampler with options
+   rs, err := NewRemoteSampler(ctx, "service_name", "cloud_platform", WithEndpoint(endpoint), WithSamplingRulesPollingInterval(350 * time.Second)); if err != nil {
```
Comment (Aneurysm9) [inline src/docs/getting-started/go-sdk/trace-manual-instr.mdx:208]: What values should be used for "cloud_platform" here?  Are those documented on the X-Ray side? Should they be included or linked here?

Comment (bhautikpip) [reply src/docs/getting-started/go-sdk/trace-manual-instr.mdx:208]: good call out. We should document equivalent value of `serviceName` and `cloudPlatform` on X-Ray side.

**Review by alolita — APPROVED** [review]
lgtm. @willarmiros lmk when ready to publish.

**Review by Aneurysm9 — APPROVED** [review]
